src/main/scala/com/sourcegraph/sbtsourcegraph/Versions.scala

```diff
@@ -67,6 +68,43 @@ object Versions {
   }
 
   private val jvmVersionCache = collection.mutable.Map.empty[Option[File], Int]
+  private var printJavaVersionPath = Option.empty[Path]
+  private def printJavaVersionFolder = {
+    def create = {
+      val dir =
+        Files.createTempDirectory("print-java-version")
+      val file = dir.resolve("PrintJavaVersion.class")
+      val base64 = scala.io.Source
+        .fromInputStream(
+          getClass()
+            .getResourceAsStream(
+              "/sbt-sourcegraph/PrintJavaVersion.class.base64"
+            )
+        )
+        .mkString
+        .trim
+
+      val contents = java.util.Base64.getDecoder().decode(base64)
+
+      Files.write(file, contents)
+      dir
+    }
+
+    printJavaVersionPath.synchronized {
+      printJavaVersionPath match {
+
+        case Some(value)
+            if value.resolve("PrintJavaVersion.class").toFile.isFile =>
+          value
+        case _ =>
+          val created = create
+          printJavaVersionPath = Some(created)
+          created
+
+      }
+    }
+
+  }
```

Comment:
This is probably unnecessarily defensive, but disappearing temp files are not unheard of.